refactor: update `blas/ext/base/gcusum` to follow current project conventions

headlessNode opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   update the package to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
refactor: update to follow current project conventions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
docs: apply code review suggestion

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
docs: apply code review suggestion

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
Merge branch 'stdlib-js:develop' into gcusum-refactor

kgryte reviewed (changes requested):
Looks like this PR needs similar changes to https://github.com/stdlib-js/stdlib/pull/4412. Would you be willing to update, @headlessNode?

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/README.md:
```diff
@@ -152,20 +141,14 @@ gcusum.ndarray( N, 0.0, x, 2, 1, y, -1, y.length-1 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = round( randu()*100.0 );
-}
+var x = discreteUniform( 10.0, -100, 100, {
```
```suggestion
var x = discreteUniform( 10, -100, 100, {
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/README.md:
```diff
@@ -116,17 +109,13 @@ The function has the following additional parameters:
 -   **offsetY**: starting index for `y`.
 
-While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, `offsetX` and `offsetY` parameters support indexing semantics based on a starting indices. For example, to calculate the cumulative sum of every other value in `x` starting from the second value and to store in the last `N` elements of `y` starting from the last element
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, offsetX and offsetY parameters support indexing semantics based on starting indices. For example, to calculate the cumulative sum of every other element in the strided input array starting from the second element and to store in the last `N` elements of the strided output array starting from the las ...
```
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, offset parameters support indexing semantics based on starting indices. For example, to calculate the cumulative sum of every other element in the strided input array starting from the second element and to store in the last `N` elements of the strided output array starting from the last element:
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/docs/repl.txt:
```diff
@@ -43,31 +43,30 @@
     While typed array views mandate a view offset based on the underlying
-    buffer, the `offset` parameter supports indexing semantics based on a
-    starting index.
+    buffer, the offset parameter support indexing semantics based on starting
```
```suggestion
    buffer, the offset parameters support indexing semantics based on starting
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/examples/index.js:
```diff
@@ -18,20 +18,14 @@
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = round( randu()*100.0 );
-}
+var x = discreteUniform( 10.0, -100, 100, {
```
```suggestion
var x = discreteUniform( 10, -100, 100, {
```

headlessNode committed:
fix: apply review suggestions

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: apply review suggestions

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: apply review suggestions

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: apply review suggestions

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
fix: apply code review suggestion

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/benchmark/benchmark.js:
```diff
@@ -21,14 +21,22 @@
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
 var gfill = require( '@stdlib/blas/ext/base/gfill' );
+var Float64Array = require( '@stdlib/array/float64' );
```
```suggestion
var zeros = require( '@stdlib/array/zeros' );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/benchmark/benchmark.js:
```diff
@@ -21,14 +21,22 @@
+// VARIABLES //
+
+var options = {
+	'dtype': 'float64'
```
```suggestion
	'dtype': 'generic'
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/benchmark/benchmark.js:
```diff
@@ -39,16 +47,8 @@ var gcusum = require( './../lib/main.js' );
-		y.push( 0.0 );
-	}
+	var x = uniform( len, -100, 100, options );
+	var y = new Float64Array( len );
```
```suggestion
	var y = zeros( len, options.dtype );
```

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/benchmark/benchmark.ndarray.js:
```diff
@@ -21,14 +21,22 @@
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
 var gfill = require( '@stdlib/blas/ext/base/gfill' );
+var Float64Array = require( '@stdlib/array/float64' );
```
Same changes to this file.

headlessNode replied:
```suggestion
var zeros = require( '@stdlib/array/zeros' );
```

headlessNode committed:
refactor: apply suggestions from code review

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/benchmark/benchmark.ndarray.js:
```diff
@@ -21,14 +21,22 @@
+// VARIABLES //
+
+var options = {
+	'dtype': 'float64'
```
```suggestion
	'dtype': 'generic'
```

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/gcusum/benchmark/benchmark.ndarray.js:
```diff
@@ -39,16 +47,8 @@ var gcusum = require( './../lib/ndarray.js' );
-		y.push( 0.0 );
-	}
+	var x = uniform( len, -100, 100, options );
+	var y = new Float64Array( len );
```
```suggestion
	var y = zeros( len, options.dtype );
```

headlessNode committed:
refactor: apply review suggestion

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented:
@kgryte changes applied.

kgryte reviewed (approved):
LGTM
